### ISSUE TITLE
[BUG]: Problems with using previous dataset splits

### ISSUE THREAD

**Issue by georgkempf** [chemprop/chemprop#310]
I wanted to train other models with sklearn_train using the same train, val, test indices from a previous training (with chemprop_train) where I used --save_smiles_splits option (which seems to write out the split_indices.pckl file for each fold). For training the second model, I used the following command

`sklearn_train --data_path ../../../dataset.csv  --model_type random_forest --dataset_type classification --split_type index_predetermined --crossval_index_file ../nn/fold_0/split_indices.pckl `

This produced the error `'Split indices must have three splits: train, validation, and test'`

I investigated a bit and it seems that a list containing the splits from different folds is expected with shape=n,3,n however the data loaded from my split_indices.pckl file has only shape=3,n i.e. [[train_indices], [val_indices], [test_indices]]. I adapted the script in this [line ](https://github.com/chemprop/chemprop/blob/359e6600405b203196ad90b24ccf694f42676845/chemprop/data/utils.py#L531 ), removing the list slice, but then ran into the problem that there are more indices loaded from the split_indices file than available in the data array causing out of index error. 

My general question is if I am using the correct approach? I am also wondering if it is possible to automatically use the split indices from all folds from the previous training for a new run with the same number of folds?

I also tried a different approach using `--data_path`, `--separate_test_path` , `--separate_val_path` options in combination with the csv files "train_full.csv", "test_full.csv", "val_full.csv" contained in the fold_N folders from the previous training but then ran into the following error:
`Reshape your data either using array.reshape(-1, 1) if your data has a single feature or array.reshape(1, -1) if it contains a single sample.`

Thanks for your help!

**Comment by kevingreenman** [comment]
Thanks for asking this question; it appears you've identified two bugs.

I can reproduce your first error message with one of the test files using
```
chemprop_train --data_path tests/data/classification_common.csv --target_columns NR-AhR --dataset_type classification --save_dir debug-results --save_smiles_splits --epochs 3 --num_folds 3
```
followed by 
```
sklearn_train --data_path tests/data/classification_common.csv --target_columns NR-AhR --dataset_type classification --save_dir debug-results --model_type random_forest --split_type index_predetermined --crossval_index_file debug-results/fold_0/split_indices.pckl
```
However, this issue is apparently not unique to sklearn, as I get the same error when I run another chemprop classification as the 2nd step:
```
chemprop_train --data_path tests/data/classification_common.csv --target_columns NR-AhR --dataset_type classification --save_dir debug-results --save_smiles_splits --epochs 3 --split_type index_predetermined --crossval_index_file debug-results/fold_0/split_indices.pckl 
```
So the first error is with the `.pckl` file.

I can reproduce your second error message with the following second step:
```
sklearn_train --data_path debug-results/fold_0/train_full.csv --separate_val_path debug-results/fold_0/val_full.csv --separate_test_path debug-results/fold_0/test_full.csv --dataset_type classification --save_dir debug-results --model_type random_forest 
```

We'll work on fixes for these issues.

**Comment by GemaRG96** [comment]
Hi! Did you find any workaround in order to train models with the same folds and splits of the data?

**Comment by kevingreenman** [comment]
We are closing all bug reports that we don't have time to fix before the release of v1.7 this month, and unfortunately this is one of them. We will soon release [Chemprop v2.0](https://github.com/chemprop/chemprop/issues/517) and do not plan to maintain/support v1.x any longer. If anyone has time to open a PR that fixes this bug before the end of January 2024, we will review it and consider including it in v1.7.

**Comment by kevingreenman** [comment]
Posting this for posterity who come searching for help with the proper format for `--crossval_index_file`:

There are some old scripts in the `scripts/` directory of the repo that I think answer this question for us: 
* https://github.com/chemprop/chemprop/blob/master/scripts/create_crossval_splits.py
* https://github.com/chemprop/chemprop/blob/master/scripts/create_crossval_index_files.py
